app/lib/actions/katello/repository/check_matching_content.rb

```diff
@@ -0,0 +1,45 @@
+module Actions
+  module Katello
+    module Repository
+      class CheckMatchingContent < Actions::Base
+        input_format do
+          param :source_repo
+          param :target_repo
+          param :contents_changed
+        end
+
+        # Check if content in the repositories has changed. We can use
+        # this info to skip regenerating metadata in pulp, keeping the
+        # revision number on the repo the same
+        def run
+          source_repo = ::Katello::Repository.find(input[:source_repo_id])
+          target_repo = ::Katello::Repository.find(input[:target_repo_id])
+
+          rpms = rpms_match?(source_repo, target_repo)
+          errata = errata_match?(source_repo, target_repo)
+          package_groups = package_groups_match?(source_repo, target_repo)
```

Comment:
Should we be checking for distributions?  If not, should we skip CheckMatchingContent on repositories that have distributions to ensure that metadata is always regenerated?